test/e2e/run_volume_test.go

```diff
@@ -145,6 +145,19 @@ var _ = Describe("Podman run with volumes", func() {
 		Expect(session).To(ExitWithError(125, fmt.Sprintf("%s: duplicate mount destination", dest)))
 	})
 
+	It("podman run with single character volume", func() {
+		// 1. create single character volume
+		session := podmanTest.Podman([]string{"volume", "create", "a"})
+		session.WaitWithDefaultTimeout()
+		volName := session.OutputToString()
+		Expect(session).Should(ExitCleanly())
+
+		// 2. create container with volume
+		session = podmanTest.Podman([]string{"run", "--volume", volName + ":/data", ALPINE, "sh", "-c", "echo hello world"})
+		session.WaitWithDefaultTimeout()
+		Expect(session).Should(ExitCleanly())
+	})
```

Comment:
This test is not actually testing your change, in particular the new code path is never executed on normal linux only when run inside the podman WSL machine because `shouldResolveWinPaths()`.

While this test does not hurt and covers the normal linux flow it is not testing your particular bug/code change, this must be tested in the machine test suite pkg/machine/e2e instead.

Reply:
I added a test in `pkg/machine/e2e/basic_test.go`